src/test/java/jp/co/yahoo/yosegi/util/TestIndexAndObject.java

```diff
@@ -0,0 +1,172 @@
+  }
+
+  @Test
+  public void T_add_get_4(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.get(5) , "test0" );
+    assertEquals( indexAndObj.get(6) , "test1" );
+  }
+
+  @Test
+  public void T_add_get_5(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    assertThrows( ArrayIndexOutOfBoundsException.class ,
+      () -> {
+        indexAndObj.get(4);
+      }
+    );
+  }
+
+  @Test
+  public void T_add_get_6(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    assertThrows( IndexOutOfBoundsException.class ,
+      () -> {
+        indexAndObj.get(6);
+      }
+    );
+  }
+
+  @Test
+  public void T_size_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    assertEquals( indexAndObj.size() , 0 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.size() , 1 );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.size() , 2 );
+  }
+
+  @Test
+  public void T_size_2(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    assertEquals( indexAndObj.size() , 0 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.size() , 1 );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.size() , 2 );
+  }
+
+  @Test
+  public void T_getNextIndex_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    assertEquals( indexAndObj.getNextIndex() , 0 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.getNextIndex() , 1 );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.getNextIndex() , 2 );
+  }
+
+  @Test
+  public void T_getNextIndex_2(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    assertEquals( indexAndObj.getNextIndex() , 5 );
+    indexAndObj.add( "test0" );
+    assertEquals( indexAndObj.getNextIndex() , 6 );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.getNextIndex() , 7 );
+  }
+
+  @Test
+  public void T_hasIndex_1(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 0 );
+    indexAndObj.add( "test0" );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.hasIndex( -1 ) , -1 );
+    assertEquals( indexAndObj.hasIndex( 0 ) , 0 );
+    assertEquals( indexAndObj.hasIndex( 1 ) , 0 );
+    assertEquals( indexAndObj.hasIndex( 2 ) , 1 );
+  }
+
+  @Test
+  public void T_hasIndex_2(){
+    IndexAndObject<String> indexAndObj = new IndexAndObject<String>( 5 );
+    indexAndObj.add( "test0" );
+    indexAndObj.add( "test1" );
+    assertEquals( indexAndObj.hasIndex( 4 ) , -1 );
+    assertEquals( indexAndObj.hasIndex( 5 ) , 0 );
+    assertEquals( indexAndObj.hasIndex( 6 ) , 0 );
+    assertEquals( indexAndObj.hasIndex( 7 ) , 1 );
+  }
+
```

Comment:
Please describe what are tested in the test method separately at T_hasIndex_?.
For example, T_hasIndex_when_xxx or some.

Reply:
Please add a prefix or suffix like T_ or Text or some so that the code reader knows it is a test method including assertThat or assertThrows and/or some testing methods.